xgcm/autogenerate.py

```diff
@@ -224,6 +234,13 @@ def position_to_relative(pos_from, pos_to):
     return to
 
 
+def raw_interp(data_left, data_right):
+    # TODO: This is not great, but I am not sure how to pass the nested
+    # function in xgcm.grid.Axis.interp to Axis._neighbor_binary_func_raw
+    # instead.
+    return 0.5*(data_left + data_right)
```

Comment:
You are correct that you can't extract that function because it is defined within a another. But there is a simple solution: define it at the top level of the grid module namespace and then use the same function in your routine and `interp`.

What I mean is to just define
```python
def interp_raw(data_left, data_right):
    # linear, centered interpolation
    # TODO: generalize to higher order interpolation
    return 0.5*(data_left + data_right)

def diff_raw ...
```
in grid.py, change the `diff` and `interp` methods to call these functions, and then import the same `_raw` functions in your module.

Does this make sense to you?